File: 20220727/모범답안/20220727_10.js/CombineTwoDigitOddNumber.js
/**
 * @param array {number[]}
 * @return {number}
 */
function solution(array) {
  let set = new Set()

  for (let i = 0; i < array.length; i++) {
    for (let j = 0; j < array.length; j++) {
      if (i === j) continue

      const num = createNumber(array[i], array[j])
      const isEven = num % 2 === 1
      if (isEven) {
        set.add(num)
      }
    }
  }

  return set.size
}

function createNumber(tenDigits, oneDigit) {
  return tenDigits * 10 + oneDigit
}

export default solution

File: 20220727/모범답안/20220727_10.js/CombineTwoDigitOddNumber.py
def solution(array):
    """
    :param array: int[]
    :return: int
    """

    mySet = set()

    for i in range(len(array)):
        for j in range(len(array)):
            if i == j:
                continue

            num = createNumber(array[i], array[j])
            isEven = num % 2 == 1
            if isEven:
                mySet.add(num)

    return len(mySet)


def createNumber(tenDigits, oneDigit):
    return tenDigits * 10 + oneDigit
